### DISCUSSION TITLE
prisma generate production issue

### DISCUSSION THREAD

**Discussion by akinyemi2161** [prisma/prisma#22763] [Q&A]
### Question

How do I fix the error below? 
Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.

I am using 2 databases which means I have 2 different schema.prisma (schema1.prisma and schema2.prisma)
even after running this script, "npx prisma generate --schema prisma/schema1.prisma && npx prisma generate --schema prisma/schema2.prisma"
I still get same error message.

### Expected behavior (optional)

Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @akinyemi2161 👋 

Thank you for raising this question.

How are you specifying the output path for both schema files? Can you share a miminal reproduction in a git repository so we can investigate?

**Reply by bamidele2161** [reply]
Hi @ludralph 
Thanks for your response 

please check this git repository
https://github.com/bamidele2161/Prisma-Fix

Thanks.

**Reply by ludralph** [reply]
Hi @akinyemi2161, 

Looking at the repository you shared, I can see that the Prisma Client generation was successful. Take a look at this [section](https://github.com/bamidele2161/Prisma-Fix/tree/master/prisma/generated) you will see the generated files there. Also running the same command `npx prisma generate --schema prisma/schema1.prisma && npx prisma generate --schema prisma/schema2.prisma`, I was able to generate the Prisma Client for both schemas as well.

**Reply by bamidele2161** [reply]
Hi @ludralph, 

Yes.
it worked fine too when I tried it locally.
But when I tried to deploy the app on AWS, I got same error

Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.

**Reply by ludralph** [reply]
Now I understand.  You did not mention you were deploying on AWS in your original message. To deploy on AWS, you would need to follow these steps:

1. Ensure that `prisma generate` is run during the build process of your AWS deployment. You can add the `prisma generate` command to the `postinstall` script in your package.json. This will make sure that Prisma Client is generated after your dependencies are installed but before your application starts.

2. If you're using a CI/CD pipeline, such as AWS Amplify, make sure that the `schema.prisma` file is included in the deployment package. If the file is missing, `prisma generate` will fail because it cannot find the schema to generate the client.

Do take a look at this [section](https://www.prisma.io/docs/orm/prisma-client/deployment/serverless/deploy-to-aws-lambda) of the documentation. This [section](https://www.prisma.io/docs/orm/prisma-client/deployment/caveats-when-deploying-to-aws-platforms) also outlines the common issues you can face when deploying to different AWS platforms

**Reply by bamidele2161** [reply]
@ludralph 
I'm encountering the issue with AWS App Runner while deploying my application that utilizes two databases. The problem arises because the schema.prisma file is missing, and the npx prisma generate command, which is targeted at schema.prisma, is unable to locate it. Instead, I'm using separate schema files named schema1.prisma and schema2.prisma.

During deployment, I've included the following script to generate Prisma client code:
npx prisma generate --schema prisma/schema1.prisma && npx prisma generate --schema prisma/schema2.prisma
However, this setup is still resulting in an error.

**Reply by bamidele2161** [reply]
@ludralph 
working fine now

thanks

**Reply by ludralph** [reply]
I'm glad to hear you got it working, may I ask what fixed it for you so others who encounter the same issue can easily make reference to it

**Reply by massimokris** [reply]
@bamidele2161 i'm facing a similar issue, how did you end solving it?

**Reply by bamidele2161** [reply]
> @bamidele2161 i'm facing a similar issue, how did you end solving it?

Hi @massimokris  fixed now?

**Reply by massimokris** [reply]
yep, i just added this directly in the dockerfile. not sure if it's the best implementation, but it works

```
COPY prisma ./prisma/
RUN npx prisma generate
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
